jpsreport/src/Analysis.cpp

```diff
@@ -0,0 +1,486 @@
+     }
+
+     if(args->GetIsMethodD()) {
+          _DoesUseMethodD = true;
+          vector<int> Measurement_Area_IDs = args->GetAreaIDforMethodD();
+          for(unsigned int i=0; i<Measurement_Area_IDs.size(); i++)
+          {
+               _areaForMethod_D.push_back(dynamic_cast<MeasurementArea_B*>( args->GetMeasurementArea(Measurement_Area_IDs[i])));
+          }
+          _StartFramesMethodD = args->GetStartFramesMethodD();
+          _StopFramesMethodD = args->GetStopFramesMethodD();
+          _IndividualFDFlags = args->GetIndividualFDFlags();
+          _geoPolyMethodD = ReadGeometry(args->GetGeometryFilename(), _areaForMethod_D);
+
+     }
+     if(args->GetIsMethodI()) {
+          _DoesUseMethodI = true;
+          vector<int> Measurement_Area_IDs = args->GetAreaIDforMethodI();
+          for(unsigned int i=0; i<Measurement_Area_IDs.size(); i++)
+          {
+               _areaForMethod_I.push_back(dynamic_cast<MeasurementArea_B*>( args->GetMeasurementArea(Measurement_Area_IDs[i])));
+          }
+          _StartFramesMethodI = args->GetStartFramesMethodI();
+          _StopFramesMethodI = args->GetStopFramesMethodI();
+          _IndividualFDFlags = args->GetIndividualFDFlags();
+          _geoPolyMethodI = ReadGeometry(args->GetGeometryFilename(), _areaForMethod_I);
+     }
+
+     _deltaF = args->GetDelatT_Vins();
+     _cutByCircle = args->GetIsCutByCircle();
+     _getProfile = args->GetIsGetProfile();
+     _isOneDimensional=args->GetIsOneDimensional();
+     _vComponent = args->GetVComponent();
+     _IgnoreBackwardMovement =args->GetIgnoreBackwardMovement();
+     _grid_size_X = int(args->GetGridSizeX());
+     _grid_size_Y = int(args->GetGridSizeY());
+     _geometryFileName=args->GetGeometryFilename();
+     _projectRootDir=args->GetProjectRootDir();
+     _trajFormat=args->GetFileFormat();
+     _cutRadius=args->GetCutRadius();
+     _circleEdges=args->GetCircleEdges();
+     _outputLocation=args->GetOutputLocation();
+}
+
+
+std::map<int, polygon_2d> Analysis::ReadGeometry(const fs::path& geometryFile, const std::vector<MeasurementArea_B*>& areas)
+{
+     Log->Write("INFO:\tReadGeometry with %s", geometryFile.string().c_str());
+     double geo_minX  = FLT_MAX;
+     double geo_minY  = FLT_MAX;
+     double geo_maxX  = -FLT_MAX;
+     double geo_maxY  = -FLT_MAX;
+
+     std::map<int, polygon_2d> geoPoly;
+
+     //loop over all areas
+     for(auto&& area: areas)
+     {
+          //search for the subroom that contains that area
+          for (auto&& it_room : _building->GetAllRooms())
+          {
+               for (auto&& it_sub : it_room.second->GetAllSubRooms())
+               {
+                    SubRoom* subroom = it_sub.second.get();
+                    point_2d point(0,0);
+                    boost::geometry::centroid(area->_poly,point);
+                    //check if the area is contained in the obstacle
+                    if(subroom->IsInSubRoom(Point(point.x()/M2CM,point.y()/M2CM)))
```

Comment:
if(subroom->IsInSubRoomOld(Point(point.x()/M2CM,point.y()/M2CM)))

Reply:
in `libcore/geometry/SubRoom.cpp` use `bool NormalSubRoom::IsInSubRoom(const Point& ped) const` from `jpsreport` and rename it accordingly.